### ISSUE TITLE
Add C and Python binding ?

### ISSUE THREAD

**Issue by natir** [natir/nuc2bit#7]
Except for Iterator it seems very easy to create a C and Python binding with cbindgen and pyo3. (maybe @luizirber can help use for python binding ?)

**Comment by luizirber** [comment]
I would suggest the C API first, because then it can be used with pretty much any language. Maybe https://github.com/lu-zero/cargo-c is a good helper for that, since it add some niceties for making it easier to link with C code (like `pkg-config` confs).

For sourmash I used https://github.com/getsentry/milksnake, which uses `cffi` for creating Python bindings. The nice thing is that it can use the header generated by `cargo-c` and doesn't need other Rust code, like PyO3 would, but PyO3 makes nicer Python APIs by default (the milksnake approach is lower-level).

**Comment by Daniel-Liu-c0deb0t** [comment]
For now, I feel like adding other language bindings is lower on the priority list than implementing all of the features that we want. However, I've found Rust to work very well with C (I have not tested Rust + Python yet), even when writing the bindings by hand, so this is definitely possible in the future.

**Comment by natir** [comment]
I agree with you it's not the main priority but I think if we want nuc2bit will be used we need a minimal C binding.

**Comment by Daniel-Liu-c0deb0t** [comment]
Yes, we can definitely keep this on the roadmap.
